### PR TITLE
Refactor PeopleViewController to subclass BaseListViewController

### LINKED ISSUE #1463: List of users is incomplete
In one of my private repos I can’t assign an issue to a user because the list of users stops at the letter O. No more users are loaded and thus they can’t be assigned.

I realise this is probably arbitrary but we only have 47 users.

<details>
<summary>Bug Report Dump (Auto-generated)</summary>
<pre>
Version 1.16.0 (1513134612)
Device: iPhone X (iOS 11.2.2)
TestFlight: false
</pre>
</details>

### PR DESCRIPTION
**Pull request by SD10**
Resolves #1463 

This PR refactors the `PeopleViewController` to match the rest of the app architecture by using `BaseListViewController` instead of `UITableViewController`.

The change from `UITableViewCell` to `UICollectionViewCell` causes us to lose the free `.checkmark` accessoryType for the cell. So I've added a `UILabel` with a ✅ for now. 

It's WIP because I'm having trouble figuring out how to update the cell for a selected/deselected state using `ListSingleSectionController`. 

ListSingleSectionControllerDelegate gives me a method:
```Swift
func didSelect(_ sectionController: ListSingleSectionController, with object: Any)
```

But I'm not familiar with IGListKit and I can't think of any way to update the cell without the `Int` for the section. 

Very likely I'm missing something.

### CODE REVIEW

File: Classes/People/PeopleViewController.swift [hunk 1]
```diff
@@ -56,83 +80,95 @@ final class PeopleViewController: UITableViewController {
...
             }
+            update(page: next as NSNumber?, animated: trueUnlessReduceMotionEnabled)
+        case .error:
+            ToastManager.showGenericError()
         }
     }
 
     @IBAction func onDone(_ sender: Any) {
         let selections = users.filter { self.selections.contains($0.login) }
+        for selection in selections { print(selection.login) }
```
Comment (SD10) [inline Classes/People/PeopleViewController.swift]: 🤦‍♂️

**Comment by rnystrom** [comment]
@SD10 we should probably use a regular section controller instead of single, that way we can put a `var selected: Boo`l on it, then when done iterate over all the section controllers who have `selected == true` to find all selected models.

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by SD10** [comment]
@rnystrom, I refactored to `PeopleSectionController`. Let me know if that is what you had in mind. 

1. I left the `ListSingleSectionController` for the empty state. I don't know if this can function without the delegate but if not I can write a boilerplate `EmptyPeopleSectionController`.

2. We need to choose a checkmark style ✔️ or ✅. Current code uses ✔️ 

githawk_ui_1024

**Comment by rnystrom** [comment]
@SD10 can we use the checkmark asset already in the assets bundle?

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by SD10** [comment]
Sure @rnystrom, updated to use the checkmark asset. Missed that. 

Trying to find my way around here!
mole

**Comment by SD10** [comment]
@rnystrom Do you want me to hold off on refactoring the milestones and label controllers until this is merged? I just want to make sure I'm not doing something incorrectly before continuing

**Comment by rnystrom** [comment]
@SD10 ya let's hold on that for the moment unless you've got some momentum built up. I want to save from shipping this in 1.18 (will land and send a 1.19 TF build ASAP) to minimize unintended breakages. That'll give this thing plenty of time to soak.

**Comment by rnystrom** [comment]
I probably broke some stuff landing that big networking refactor.

**Comment by SD10** [comment]
No problem at all, can fix this tomorrow

**Comment by SD10** [comment]
Sorry, took awhile to get back to this. I think I've resolved the conflicts but have no time to test the changes tonight

File: Classes/People/PeopleViewController.swift [hunk 2]
```diff
@@ -58,83 +81,88 @@ final class PeopleViewController: UITableViewController {
...
+                    self?.users = sortedUsers
+                }
+                self?.update(page: response.next as NSNumber?, animated: trueUnlessReduceMotionEnabled)
             case .failure:
                 ToastManager.showGenericError()
             }
         }
     }
 
+    func handle(result: Result<V3DataResponse<[V3User]>>, append: Bool) {
```
Comment (SD10) [inline Classes/People/PeopleViewController.swift]: Oops 🤦‍♂️

**Comment by rnystrom** [comment]
@SD10 lmk what you think here. Now that #1633 and #1632 are merged I'd love to explore making the labels and milestones UIs context menus.

**Comment by SD10** [comment]
@rnystrom I was testing this now actually. So sorry about the delays 😓 This control would not be a context menu though, correct?

**Comment by rnystrom** [comment]
@SD10 np! In no hurry. I actually think this one could be a good candidate too, just scrollable.

**Comment by SD10** [comment]
Nevermind, I understand now. Before I had made the assumption that this controller was also used on top of the `MessageTextView`. Now I realize they're not the same controller but likely just both had paging broken.

**Comment by SD10** [comment]
@rnystrom I actually already refactored the labels and milestone controllers as well 🤣 I just didn't submit it because I was waiting on this. Want me to explore making all 3 the context menus?

**Comment by rnystrom** [comment]
@SD10 how bout this:

- Once you're g2g on this PR, lmk and I'll review + land
- Assuming labels + milestones follow the same pattern, just put up another PR w/ both and I'll #yolo land
- Then we can followup making them context menus. There's a chance this is a little more complicated b/c of changing UX, lib might be missing features, etc.

Sound good?

**Comment by SD10** [comment]
**an hour later...*

I honestly can't figure out how to get GitHawk to open in the simulator 😂 I keep getting this error after Safari login/authorization

screen shot 2018-03-11 at 4 03 24 pm

**Comment by rnystrom** [comment]
@SD10 you might want to rebase again?

**Pull request merged**